src/excmds.ts

```diff
@@ -3281,32 +3281,78 @@ export async function yankimage(url: string): Promise<void> {
+    @param id
+        A bare number means the current window is used. Starts at 1. 0 refers to last tab of the current window, -1 to penultimate tab, etc.
+
+        A string following the following format: "[0-9]+.[0-9]+" means the first number being the index of the window that should be selected and the second one being the index of the tab within that window. [[taball]] has completions for this format.
 
         "#" means the tab that was last accessed in this window
 
-    This is different from [[taball]] because `index` is the position of the tab in the current window.
+        A non integer string means to search the URL and title for matches, in this window if called from tab, all windows if called from anytab. Title matches can contain '*' as a wildcard.
  */
 //#background
-export async function tab(index: number | "#") {
-    return tabIndexSetActive(index)
+export async function tab(...id: string[]) {
+    return tab_helper(true, false, ...id)
 }
 
-/** Change active tab.
+/** Wrapper for [[tab]] with multi-window completions
+ */
+//#background
+export async function taball(...id: string[]) {
+    return tab_helper(true, true, ...id);
+}
 
-    @param id
-        A string following the following format: "[0-9]+.[0-9]+", the first number being the index of the window that should be selected and the second one being the index of the tab within that window.
+/** Helper to change active tab. Used by [[tab]] and [[taball]].
 
+    @param interactive
+        Controls if we should prompt if multiple matches are found, or just pick the first match
+
+    @param anyWindow
+        True if we should search in all windows, or just the current one.
+
+    @param key
+        String or int tab search key, see [[tab]] for usage.
  */
 //#background
-export async function taball(id: string) {
+export async function tab_helper(interactive: boolean, anyWindow: boolean, ...key: string[]) {
+    const id = key.join(" ")
+    if (Number.isInteger(Number(id)))
+        return tabIndexSetActive(Number(id));
+    if (id === "#")
+        return tabIndexSetActive(id);
+
+    if (id !== null && id !== undefined && !/\d+\.\d+/.exec(id)) {
+        let defaultQuery = {}
+        if (!anyWindow)
+            defaultQuery = {windowId: (await activeTab()).windowId};
+
+        let results = []
+        try {
+            results = results.concat(await browser.tabs.query({...defaultQuery, ...{url: id}}))
+        } catch (e) { }
+        if (results.length < 2)
+            results = results.concat(await browser.tabs.query({...defaultQuery, ...{title: id.replace("*", "\\*")}}))
+        if (results.length < 2)
+            results = results.concat(await browser.tabs.query(defaultQuery)).filter((tab => tab.url.includes(id)))
+        if (results.length < 2)
+            results = results.concat(await browser.tabs.query({...defaultQuery, ...{title: "*" + id + "*"}}))
+        if (results.length) {
```

Comment:
In practice this feels like the filtering might be a bit too loose:

```
yarn run build && yarn run run # starts Tridactyl with the new tab page and txti.es open
:set completions.Tab.autoselect false
:tab txti # :taball txti is prefilled. I expected to be taken to the txti.es tab
```

Reply:
oops, I meant for this to be a set, hopefully this will fix it. I think `tab.id` is unique in the browser session, is that true?

Reply:
Yes. I think : )